backend/donor-service/main.js

```diff
@@ -570,7 +580,21 @@ app.post('/auth/mobile/verifyOTP', async(req, res) => {
         console.debug('OTP verified', verifyOtpResponse);
         if (R.pathOr([], ["mobileLinkedHid"], verifyOtpResponse).length > 0) {
             for (const data of verifyOtpResponse.mobileLinkedHid) {
-                data["pledged"] = await isABHARegistered(data.healthIdNumber)
+                let abhaRegistered = await isABHARegistered(data.healthIdNumber, true)
+                data["pledged"] = abhaRegistered
+                if (abhaRegistered) {
+                    let pledgeStatus = await getPledgeStatus(data.healthIdNumber)
+                    switch(pledgeStatus) {
+                        case true : 
```

Comment:
should this be PLEDGE_STATUS.PLEDGED?

Reply:
In the current logic, we are storing the pledge status value as true. Since the new logic stores the values as 0,1 or 2 and there might be old keys with the value as true in the Redis. We can write a script to make them 0. But there should be a check in the logic as well, If someone misses out on this point.

Reply:
this can bring confusions latter, since we are not in production right now,
we can write a script to convert the values in redis to respective values